service/history/task/transfer_task_executor_base.go

```diff
@@ -428,6 +458,20 @@ func appendContextHeaderToSearchAttributes(attr, context map[string][]byte, allo
 	return attr, nil
 }
 
+func getWorkflowHeaders(startEvent *types.HistoryEvent) map[string][]byte {
+	attr := startEvent.GetWorkflowExecutionStartedEventAttributes()
+	if attr == nil || attr.Header == nil {
+		return nil
+	}
+	headers := make(map[string][]byte)
```

Comment:
minor nit: this could be `make(map[string][]byte, len(attr.Header.Fields))`

Reply:
fixed